compiler/testData/codegen/bytecodeText/conditions/nullCompareSideEffectingExpression.kt

```diff
@@ -0,0 +1,16 @@
+fun f() : Int {
```

Comment:
pls, don't use println in test
it would be best to rewrite it via 
```var result = "fail"

fun f() : Int {
    result = "OK"
    return 42
}

fun box(): String {
    if (f() == null) {
        return "fail"
    }
    result = "fail"

    if (f() != null) {
        return result
    }

    return "fail"
}
```
And add same test to box ones.